### PR TITLE
#159204719 Enable social login

### PR DESCRIPTION
**Pull request by Oluwafayokemi**
## What does this Pr do?
Enable social login using Facebook and google
## Description of Task to be completed?
implement passport authentication for google
implement passport authentication for facebook
mock test for social login authentication
## How should this be manually tested?
- clone the repository, 
- create the database for test and development
- create a .env file and follow the sample .env to fill in the necessary data
- run npm init
- npm install
- npm test
- Go to your local browser, 
- login to '/api/auth/google' to authenticate with google
- login to '/api/auth/facebook' to authenticate with facebook
## What are the necessary pivotal tracker stories?
[#159204719](https://www.pivotaltracker.com/story/show/159204719)
## Screenshots (if appropriate)
image

image

### CODE REVIEW

**Review by Veraclins — COMMENTED** [review]
@Oluwafayokemi nice job completing the task. You have really done well. I just have a few concerns. It seems Code climate is failing and coverage is way down. You might want to look into your tests and refactoring duplicated codes. This will increase coverage and fix the code climate issues and make your work perfect. Good job!

**Review by Veraclins — COMMENTED** [review]
@Oluwafayokemi nice job completing the task. You have really done well. I just have a few concerns. It seems Code climate is failing and coverage is way down. You might want to look into your tests and refactoring duplicated codes. This will increase coverage and fix the code climate issues and make your work perfect. Good job!

**Review by yomigeek — APPROVED** [review]

**Review by yomigeek — APPROVED** [review]
Nice job @Oluwafayokemi , I like the way the code is structured.

**Review by Veraclins — APPROVED** [review]

**Review by chukwuemekachm — APPROVED** [review]

File: server/controllers/VerifyController.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  }
+
+  /**
+   * Re-sends a verification email to the user
+   * @param {object} req the request object
+   * @param {object} res the response object
+   * @param {object} next the next middleware function
+   */
+  static resendVerificationEmail(req, res, next) {
+    // get the email from the user
```
Comment (tonymontaro) [inline server/controllers/VerifyController.js]: Good job! Let's remove un-necessary comments as the JS-doc is more than sufficient in this case.

**Pull request merged**